specification.markdown

```diff
@@ -228,6 +210,95 @@ VINT\_WIDTH | VINT\_MARKER | VINT\_DATA     | Element Data Size Status
+
+    Endianness:     Big-endian
+    Length:         A Signed Integer Element MUST at least 1 octet in
+                    length and no more than 8 octets in length.
+    Definition:     A Signed Integer stores an integer (meaning that it
+                    can be written without a fractional component) which
+                    may be negative, positive, or zero. Because EBML
+                    limits Signed Integers to 8 octets in length a
+                    Signed Element may store a number from
+                    −9,223,372,036,854,775,808 to
+                    +9,223,372,036,854,775,807.
+
+Element Data Type:   Unsigned Integer
+
+    Endianness:     Big-endian
+    Length:         An Unsigned Integer Element MUST at least 1 octet in
+                    length and no more than 8 octets in length.
+    Definition:     An Unsigned Integer stores an integer (meaning that
+                    it can be written without a fractional component)
+                    which may be positive or zero. Because EBML limits
+                    Unsigned Integers to 8 octets in length an unsigned
+                    Element may store a number from 0 to
+                    18,446,744,073,709,551,615.
+
+Element Data Type:   Float
+
+    Endianness:     Big-endian
+    Length:         A Float Element MUST either 4 octets (32 bit) or 8
+                    octets (64 bit) in length.
+    Definition:     A Float Elements stores a floating-point number as
+                    defined in IEEE 754.
+
+Element Data Type:   String
+
+    Endianness:     None
+    Length:         A String Element may use any length (included zero)
+                    up to the maximum Element Data Size value permitted.
+    Definition:     A String Element may either be empty (zero-length)
+                    or contain Printable ASCII characters in the range
+                    of 0x20 to 0x7E. Octets with all bits set to zero
+                    may follow the string value when needed.
+
+Element Data Type:   UTF-8
+
+    Endianness:     None
+    Length:         A UTF-8 Element may use any length (included zero)
+                    up to the maximum Element Data Size value permitted.
+    Definition:     A UTF-8 Element shall contain only a valid Unicode
+                    string as defined in [RFC 2279](http://www.faqs.org/rfcs/rfc2279.html).
+                    Octets with all bits set to zero may follow the
+                    UTF-8 value when needed.
+
+Element Data Type:   Date
+
+    Endianness:     None
+    Length:         A Date Element MUST be 8 octets in length.
+    Definition:     The Date Element MUST contain a Signed Integer that
+                    expresses a point in time referenced in nanoseconds
+                    from the precise beginning of the third millennium
+                    of the Gregorian Calendar in Coordinated Universal
+                    Time (also known as 2001-01-01T00:00:00.000000000
```

Comment:
EPOCH 32 bits was not precise enough. So since we needed something else, we used a different reference point to avoid confusion and easily point out errors. The beginning of the millennium seemed like a fancy date.